satpy/readers/seviri_base.py

```diff
@@ -15,1159 +14,25 @@
-
-    """
-    return Chebyshev(coefs, domain=domain)(time) - 0.5 * coefs[0]
-
-
-def chebyshev_3d(coefs, time, domain):
-    """Evaluate Chebyshev Polynomials for three dimensions (x, y, z).
-
-    Expects the three coefficient sets to be defined in the same domain.
-
-    Args:
-        coefs: (x, y, z) coefficient sets.
-        time: See :func:`chebyshev`
-        domain: See :func:`chebyshev`
-
-    Returns:
-        Polynomials evaluated in (x, y, z) dimension.
-    """
-    x_coefs, y_coefs, z_coefs = coefs
-    x = chebyshev(x_coefs, time, domain)
-    y = chebyshev(y_coefs, time, domain)
-    z = chebyshev(z_coefs, time, domain)
-    return x, y, z
-
-
-class NoValidOrbitParams(Exception):
-    """Exception when validOrbitParameters are missing."""
-
-    pass
-
-
-class OrbitPolynomial:
-    """Polynomial encoding the satellite position.
-
-    Satellite position as a function of time is encoded in the coefficients
-    of an 8th-order Chebyshev polynomial.
-    """
-
-    def __init__(self, coefs, start_time, end_time):
-        """Initialize the polynomial."""
-        self.coefs = coefs
-        self.start_time = start_time
-        self.end_time = end_time
-
-    def evaluate(self, time):
-        """Get satellite position in earth-centered cartesian coordinates.
-
-        Args:
-            time: Timestamp where to evaluate the polynomial
-
-        Returns:
-            Earth-centered cartesian coordinates (x, y, z) in meters
-        """
-        domain = [np.datetime64(self.start_time).astype("int64"),
-                  np.datetime64(self.end_time).astype("int64")]
-        time = np.datetime64(time).astype("int64")
-        x, y, z = chebyshev_3d(self.coefs, time, domain)
-        return x * 1000, y * 1000, z * 1000  # km -> m
-
-    def __eq__(self, other):
-        """Test equality of two orbit polynomials."""
-        return (
-            np.array_equal(self.coefs, np.array(other.coefs)) and
-            self.start_time == other.start_time and
-            self.end_time == other.end_time
-        )
-
-
-def get_satpos(orbit_polynomial, time, semi_major_axis, semi_minor_axis):
-    """Get satellite position in geodetic coordinates.
-
-    Args:
-        orbit_polynomial: OrbitPolynomial instance
-        time: Timestamp where to evaluate the polynomial
-        semi_major_axis: Semi-major axis of the ellipsoid
-        semi_minor_axis: Semi-minor axis of the ellipsoid
-
-    Returns:
-        Longitude [deg east], Latitude [deg north] and Altitude [m]
-    """
-    x, y, z = orbit_polynomial.evaluate(time)
-    geocent = pyproj.CRS(
-        proj="geocent", a=semi_major_axis, b=semi_minor_axis, units="m"
+    warnings.warn(
+        f"'satpy.readers.seviri_base.{name}' has been moved to 'satpy.readers.{new_submod}.{name}'. "
+        f"Import from the new location instead (ex. 'from satpy.readers.{new_submod} import {name}'). "
+        "The old import paths will be removed in Satpy 1.0",
+        stacklevel=2,
```

Comment:
The warnings in moved files look similar. Is it possible to factorize them into a helper module?

Reply:
No, I don't think so. Then the warning would point to the warning module, not the originating file.